In [88]:
%matplotlib inline
import pandas as pd
from nltk import sentiment
from IPython.display import display, Markdown
from nltk.sentiment.vader import SentimentIntensityAnalyzer
from nltk import tokenize
from typing import *

In [62]:
import nltk
nltk.download('vader_lexicon')

[nltk_data] Downloading package vader_lexicon to C:\Users\Kaden's
[nltk_data]     Laptop\AppData\Roaming\nltk_data...


True

# Sentiment_Analyzer

#### BOTH OTHER FILES STORED IN THIS REPOSITORY WERE USED AS REFERENCE FOR THIS FILES CREATION

We begin by importing the data file and looking at a portion of our data

In [47]:
data = pd.read_csv(r"C:\Users\Kaden's Laptop\Documents\GitHub\KadenFranklin.github.io\sentiment_analyzer\names1.csv")
print(data.head())
data.describe()

  Local Names                   Translation    Location Real Names
0      Abombo                            _   Kisanagani     Dupont
1       Adami                        Madam       Opala     generic
2     Adamisu  Knowing eyes, control, anger      Basoko          _
3       Aduyi                          Lion           _          _
4       Aginu                      Kneeling   Kisangani      Genot


Local Names Translation Location Real Names
count          271         270      267        264
unique         270         188       91        100
top      Tuku-Tuku           _  Katanga          _
freq             2          65       26        123

The data gathered for this analyis is collected from King Leopold's Congo and the "Scramble for Africa" by Michael A. Rutz

Specifically from page 163 and the 'Sample of names used in this word'.

We can see from the commands above that there are 271 rows of names, with 270 unique names, 188 distinct translations, originating from 91 locations. In addition to this, asside from collective 'meta' names there are 99 names given to colonials by natives.

The datatypes are as follows:

In [48]:
data.dtypes

Local Names    object
Translation    object
Location       object
Real Names     object
dtype: object

Local Name - Object

Translation - Object

Location - Object	

Real Name - Object

Now just to be certain we will check the dataset for duplicates.

In [49]:
i = -1
for x in data.duplicated():
    i += 1
    if x == True:
        print(x)
        print(i)

Success, no duplicates, Now we move on to the interesting part, [sentiment analysis](https://monkeylearn.com/sentiment-analysis/).

Now we can split our data into its corresponding columns so that we can work with it more easily.


In [50]:
names_1 = list(data["Local Names"])
transl = list(data["Translation"])
location = list(data["Location"])
names_2 = list(data['Real Names'])

The function below defines the [markdown](https://www.markdownguide.org/) table, which the results of our analysis will be stored.

In [85]:
def show_markdown_table(headers: List[str], data: List) -> str:
    s = f"| {' | '.join(headers)} |\n| {' | '.join([(max(1, len(header) - 1)) * '-' + ':' for header in headers])} |\n"
    for row in data:
        s += f"| {' | '.join([str(item) for item in row])} |\n"
    display(Markdown(s))

The function below takes in an input of either one name, or a list of names. It will return a resulting markdown table giving the scores of such input names.

In [99]:
def en_name_analyzer(name):
    dis=[]
    sid = SentimentIntensityAnalyzer()
    
    if isinstance(name, list):
        dis.append([[sid.polarity_scores(x)] for x in name])
        return dis
                
    else:
        return sid.polarity_scores(name)


In [107]:
def sw_name_analyzer(name):

    sid = SentimentIntensityAnalyzer()    

    return sid.polarity_scores(name)


Having defined such functions we can perform an analysis on our own dataset.

In [113]:
#en_name_analyzer(transl)
#en_name_analyzer(names_2)
i = -1

headers = ['local name', 'score', 'location']
rows = [[names_1[i], sw_name_analyzer(names), location[i]] for names in names_1]
show_markdown_table(headers, rows)

SyntaxError: invalid syntax (1033291091.py, line 6)

Now that we have performed the necessary evaluation of our names, lets organize the data.

In [41]:
# Organize by region here

In this file we have used the [Vader](https://towardsdatascience.com/sentimental-analysis-using-vader-a3415fef7664) library to perform Sentiment Analysis on a list of roughly 270 names. Vader allows for the performance of multilingual Sentiment analysis through 3rd party translation APIs. What this means is that inputs from other languages are translated to english in order to calculate a sentiment score. I tried looking for a Natural Language Processing database in Swahili, Kikongo, & Lingala. But resources in computer science are sparse in these langauges.

For the reasons above, the words with translated sentiment should be weighted less in the consideration of the word's lasting impact today.